content/geometry/PolygonUnionArea.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * Author: black_horse2014, chilli
+ * Date: 2019-10-29
+ * License: CC0
+ * Source:
+ * Description: Calculates the area of the union of $n$ polygons (not necessarily convex).
+ * Status: Submitted on ECNA 2017 Problem A.
+ * Time: $O(N^2)$, where $N$ is the total number of points
+ * Usage:
+ */
+#pragma once
+
+#include "Point.h"
+#include "sideOf.h"
+
+typedef Point<double> P;
+double rat2(P a, P b) { return sgn(b.x) ? a.x/b.x : a.y/b.y; }
+double polyUnion(vector<vector<P>> &poly) {
+	double ret = 0;
+	rep(i,0,sz(poly)) rep(v,0,sz(poly[i])) {
+		P A = poly[i][v], B = poly[i][(v + 1) % sz(poly[i])];
+		vector<pair<double, int>> segs({{0, 0}, {1, 0}});
+		rep(j,0,sz(poly)) {
+			if (i == j) continue;
+			rep(u,0,sz(poly[j])) {
+				P C = poly[j][u], D = poly[j][(u + 1) % sz(poly[j])];
+				int sc = sideOf(A, B, C), sd = sideOf(A, B, D);
+				if (sc != sd) {
+					double sa = C.cross(D, A), sb = C.cross(D, B);
+					if (min(sc, sd) < 0)
+						segs.emplace_back(sa / (sa - sb), sgn(sc - sd));
+				} else if (!sc && !sd && j<i && sgn((B-A).dot(D-C))>0){
```

Comment:
There's comparisons being done on doubles everywhere here, so I guess a requirement is that the coordinates all be integers < ~2^25?

Reply:
`sgn(x) > 0` is the same as `x > 0`, right? or is the idea that epsilon checks can be added to `sgn` and `sideOf`? That should be mentioned if so

Reply:
there's not actually that many comparisons on doubles. There's the `sideOf` comparisons, the comparison in `rat`, and one extra comparison in the conditional. I think that's a reasonable condition.

Yeah I'll add some notes about epsilons.